rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -27,40 +27,64 @@ namespace memory_strategies
 
 namespace static_memory_strategy
 {
-  struct ObjectPoolBounds
-  {
-  public:
-    size_t max_subscriptions_;
-    size_t max_services_;
-    size_t max_clients_;
-    size_t max_executables_;
-    size_t pool_size_;
-
-    ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
-      size_t executables = 1, size_t pool = 1024)
-    : max_subscriptions_(subs), max_services_(services), max_clients_(clients), max_executables_(
-        executables), pool_size_(pool)
-    {}
-  };
+struct ObjectPoolBounds
+{
+public:
+  size_t max_subscriptions_;
+  size_t max_services_;
+  size_t max_clients_;
+  size_t max_executables_;
+  size_t max_guard_conditions_;
+  size_t pool_size_;
+
+  ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
+    size_t executables = 1, size_t guard_conditions = 2, size_t pool = 1024)
```

Comment:
I think this is a good example where the method chaining interface (which we added to the ROS 2 messages) is a better approach. The positional arguments are almost impossible to introspect for the user.

May be we should only provide a default constructor and setters?